chainer/training/extensions/_snapshot.py

```diff
@@ -108,6 +193,15 @@ def snapshot(savefun=None,
             used.
         snapshot_on_error (bool): Whether to take a snapshot in case trainer
             loop has been failed.
+        num_retain (int): Number of snapshot files to retain
+            through the cleanup. Must be positive integer. Automatic deletion
+            of old snapshots only works when the filename is string.
+        lister (callable): A function to find files from Trainer output
+            directory.
+        loadfun (callable): A function to load file content to the target.
```

Comment:
Hmm, do you have to specify this?
As `savefun` is already given, and `savefun` and `loadfun` are essentially the two aspects of a single logic, I feel it's bizzare to have to give both...

Reply:
I understand that. This is needed for future extension as I mentioned above, but the abstraction might be unmature. I want further discussion about this, separate from this PR.

Reply:
I don't have good idea on how to obtain the loading function, as `savefun` itself doesn't indicate anything to computers...

Reply:
How about omitting `loadfun` in this PR and assuming `savefun` is `save_npz`?
Or do you want to specify something other than `load_npz`?
If so, we would have to either reconsider the design of `savefun/loadfun`, or make `loadfun` a hidden argument.

Reply:
Assuming `save_npz` would be nice for now. I'll remove it tomorrow.

Reply:
Addressed at 1f0c81e , but with no safeguard...